apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/BigtableReadRowsRequestManagerTest.cs

```diff
@@ -0,0 +1,187 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Xunit;
+
+namespace Google.Cloud.Bigtable.V2.Tests
+{
+    /// <summary>
+    /// Test for <see cref="BigtableReadRowsRequestManager"/>.
+    /// </summary>
+    public class BigtableReadRowsRequestManagerTest
+    {
+        private static ReadRowsRequest CreateRowRangeRequest(RowRange rowRange) =>
+            new ReadRowsRequest { Rows = RowSet.FromRowRanges(rowRange) };
+
+        private static ReadRowsRequest CreateRowKeysRequest(params BigtableByteString[] keys) =>
+            new ReadRowsRequest { Rows = RowSet.FromRowKeys(keys) };
+
+        private static ReadRowsRequest CreateRowFilterRequest(RowFilter rowFilter) =>
+            new ReadRowsRequest { Filter = rowFilter };
+
+        /// <summary>
+        /// Test rowfilter appears in the <see cref="BigtableReadRowsRequestManager.BuildUpdatedRequest()"/>
+        /// </summary>
+        [Fact]
+        public void TestRowFilter()
+        {
+            var rowFilter = RowFilters.Chain(
+                RowFilters.CellsPerColumnLimit(1),
+                RowFilters.CellsPerRowOffset(2),
+                RowFilters.CellsPerRowLimit(10));
+
+            ReadRowsRequest originalRequest = CreateRowFilterRequest(rowFilter);
+
+            BigtableReadRowsRequestManager underTest = new BigtableReadRowsRequestManager(originalRequest);
+
+            Assert.Equal(originalRequest, underTest.BuildUpdatedRequest());
+        }
+
+        /// <summary>
+        /// Test a single, full table scan scenario for updating <see cref="BigtableReadRowsRequestManager.LastFoundKey"/>.
+        /// </summary>
+        [Fact]
+        public void TestFilterRowsTestAllRange()
+        {
+            BigtableByteString key1 = "row1";
+
+            ReadRowsRequest originalRequest = CreateRowRangeRequest(RowRange.ClosedOpen(null, null));
+
+            BigtableReadRowsRequestManager underTest = new BigtableReadRowsRequestManager(originalRequest);
+
+            underTest.LastFoundKey = key1;
+            Assert.Equal(CreateRowRangeRequest(RowRange.Open(key1, null)), underTest.BuildUpdatedRequest());
+
+        }
+
+        /// <summary>
+        ///  Test rowKeys scenario for <see cref="BigtableReadRowsRequestManager.BuildUpdatedRequest()"/>.
+        /// </summary>
+        [Fact]
+        public void TestFilterRowsRowKeys()
+        {
+            BigtableByteString key1 = "row1";
+            BigtableByteString key2 = "row2";
+            BigtableByteString key3 = "row3";
+
+            ReadRowsRequest originalRequest = CreateRowKeysRequest(key1, key2, key3);
+
+            BigtableReadRowsRequestManager underTest = new BigtableReadRowsRequestManager(originalRequest);
+
+            Assert.Equal(originalRequest, underTest.BuildUpdatedRequest());
+            underTest.LastFoundKey = key1;
```

Comment:
I don't think so, you probably mean a case with `RowRange`(from "row01" to "row10") and `LastFoundKey` would be "row07". I will add such a test. 

When we set up `Rows.RowKeys` we are passing a list of specific `rowKeys` we are looking for, in this example "row1", "row2" and "row3". There should not be a found `rowKey` "row1a". It would actually be an equivalent of `processingStatus.Invalid` from `MutateRowsRequestManager`. 

Hmm we don't even check that `LastFoundKey` is actually within the list/range of `rowKeys` we are requesting, should we?

Reply:
Nope, I meant what I said, but yes checking for the range cut in the middle somewhere would be good too.

"There should not be a found `rowKey` "row1a" - Yes, there won't be a found key like that, but the proto field, found [here](https://github.com/evildour/googleapis/blob/master/google/bigtable/v2/bigtable.proto#L171), indicates the last key _scanned_, not found. I'm guessing if it has to skip form "row1" to "row2" internally but return a response before getting to "row2", `last_scanned_row_key` will let you know where it left off. We can check internally to confirm, but it wouldn't hurt to have the test anyway just in case the server does return a last scanned key that doesn't match one being searched for.

Reply:
Added the test